[WIP] OSDOCS-2210: OSDK-1713 Custom Scorecard storage feature (#36751)
openshift/openshift-docs: modules/osdk-scorecard-store-output.adoc

// Module included in the following assemblies:
//
// * operators/operator_sdk/osdk-scorecard.adoc

:osdk_ver: v1.8.0

[id="osdk-scorecard-store-output_{context}"]

> = Scorecard output storage

Reviewer: Do we need to add more context about this feature?  Feels like there should be more. I'll have @theishshah look this over as well.

Author: @jmrodri @theishshah I am happy to add additional context. I was working primarily off of the [upstream docs ](https://sdk.operatorframework.io/docs/testing-operators/scorecard/custom-tests/#storing-scorecard-test-output), so please let me know what I can add.

Reviewer 2: This feature can also be enabled in the config.yaml via the following:
```
tests:
  - entrypoint:
    - scorecard-test
    - <test-entrypoint>
    image: quay.io/operator-framework/scorecard-test:v1.10.0
    labels:
      suite: <test-suite>
      test: <test-name>
    storage:      
      spec:        
        mountPath:
          path: /test-output
```

Additionally if we can add a line at the beginning stating that users can enable local storage of scorecard results in either the CLI or in the config files.

Thanks.

Author: @theishshah I have made the requested changes along with a little bit of clean up. Would you PTAL?

// Module included in the following assemblies:
//
// * operators/operator_sdk/osdk-scorecard.adoc

:osdk_ver: v1.16.0

> :_content-type: MODULE

Reviewer 3: Please pick a valid [content type](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#content-type-attributes).

[id="osdk-scorecard-store-output_{context}"]
= Local storage of scorecard results

> You can enable local storage of scorecard test results by using the `scorecard` command or by modifying your `config.yaml` file.

Reviewer 3: This sounds like the abstract for a procedure module. I strongly suggest refactoring the content into discrete steps that follow the procedure template.

[IMPORTANT]
====
There is a known issue with the Operator SDK scorecard storage utility. Currently, running the `operator-sdk scorecard` command with the `--test-output` flag returns an empty directory. There is no known workaround at this time. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2017669[*BZ#2017669*])
====

Running the `scorecard` command with the `--test-output` flag stores the scorecard test results in a specified directory. If you do not specify a directory, the command stores the test results in the `test-output` directory of your working directory.

.`scorecard --test-output` command syntax
[source,terminal,subs="attributes+"]
----
$ operator-sdk scorecard <bundle_directory_or_image> --test-output=/<output_directory> <.>
----

> <.> Define the output directory.

Reviewer 3: Honestly, I'd rewrite this whole thing without the broken flag and be like "this command store the test results in the `test-output` directory of your working directory." When the bug is fixed, update the docs to describe the flag.

Reviewer 4: +1 The why this all is written makes this read as if 'we have this cool new feature, but none of it works - good luck'. 

I wouldn't document this till the bug was/is fixed; only document the way's in which it properly works.

> To enable local storage in your configuration files, you add the following to your `config.yaml` file:

Reviewer 3: ```suggestion
To enable local storage in your configuration files, you add the following <noun_goes_here> to your `config.yaml` file:
```

.Example scorecard configuration file
[source,terminal,subs="attributes+"]
----
...

> tests:

Reviewer 3: Does it matter where you place this stanza?

  - entrypoint:
    - scorecard-test
    - <test_entrypoint> <.>
    image: quay.io/operator-framework/scorecard-test:v1.10.0
    labels:
      suite: <test_suite> <.>
      test: <test_name> <.>
    storage:
      spec:
        mountPath:
          path: /<output_directory> <.>
----

> <.> Replace with the command and arguments that you want to call, such as `scorecared-test`.

Reviewer 3: ```suggestion
<.> Replace `<test_entrypoint>` with the command and arguments to run, such as `scorecared-test`.
```
Will the user have enough context to define all of these parameters? If not, you might want to circle back and add more context.

> <.> Define the test suite that you want to run, such as `basic` or `olm`.

Reviewer 3: Are these the only two options?

<.> Define the test name that you want to run, such as `olm-bundle-validation-test`.

> <.> Set the directory where you want to store your test results, such as `test-results`.

Reviewer 3: ```suggestion
<.> Set the location of the directory to store your test results, such as `test-results`.
```
Does this directory have to exist?
